### PR TITLE
Feat:rdcc 5419 new

### PR DESCRIPTION
**Pull request by shivajihmcts**
### JIRA link ###

https://tools.hmcts.net/jira/browse/RDCC-5419

### Change description ###

Staff: Create New API to retrieve Skills for users

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by SujeethKumarAri — APPROVED** [review]

**Review by sahitya-desireddy — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        assertThat(HttpStatus.FORBIDDEN.value()).isEqualTo(fetchResponse.statusCode());
+        assertThat(fetchResponse.getBody().asString()).contains(getToggledOffMessage());
+
+    }
+
+    @Test
+    @ToggleEnable(mapKey = STAFF_REF_DATA_RD_STAFF_UI_KEY, withFeature = true)
+    @ExtendWith(FeatureToggleConditionExtension.class)
+    public void should_return_401_when_authentication_invalid() {
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java:75]: can you add another test case for non staff admin user

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        assertThat(serviceSkill.getId()).isEqualTo("1");
+
+        SkillDTO skillDTO = serviceSkill.getSkills().get(0);
+
+        assertThat(skillDTO.getSkillId()).isEqualTo(1L);
+        assertThat(skillDTO.getSkillCode()).isEqualTo("1");
+        assertThat(skillDTO.getDescription()).isEqualTo("testskill1");
+        assertThat(skillDTO.getUserType()).isEqualTo("1");
+    }
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataIntegrationTest.java:62]: can you add another test case for empty skills

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java [hunk 3]
```diff
@@ -41,9 +43,51 @@ public class StaffRefDataController {
...
+                    code = 200,
+                    message = "Successfully retrieved list of ServiceSkills for the request provided",
+                    response = StaffWorkerSkillResponse.class
+            ),
+            @ApiResponse(
+                    code = 400,
+                    message = "Bad Request"
+            ),
+            @ApiResponse(
+                    code = 401,
```
Comment (rajkatla-hmcts) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java:70]: is 401 status code correct ?

File: src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java [hunk 4]
```diff
@@ -60,17 +70,50 @@ void setUp() {
         MockitoAnnotations.openMocks(this);
     }
 
+    @Test
+    void should_return_service_skills_with_status_code_200() {
+        StaffWorkerSkillResponse staffWorkerSkillResponse =
```
Comment (rajkatla-hmcts) [inline src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java:75]: can add couple of skills to staffWorkerSkillResponse and then assert the size and content

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import static uk.gov.hmcts.reform.cwrdapi.util.FeatureToggleConditionExtension.getToggledOffMessage;
+
+@ComponentScan("uk.gov.hmcts.reform.cwrdapi")
+@WithTags({@WithTag("testType:Functional")})
+@ActiveProfiles("functional")
+@SerenityTest
+@SpringBootTest
+@Slf4j
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class StaffRefDataFunctionalTest extends AuthorizationFunctionalTest {
```
Comment (rajkatla-hmcts) [inline src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java:30]: and also change the class name to StaffRefDataSkillsFunctionalTest

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java (hunk 4 above)
Comment (shivajihmcts) [reply src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java:75]: done

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java (hunk 5 above)
Comment (shivajihmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java (hunk 4 above)
Comment (rajkatla-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java:75]: @shivajihmcts i dont see new changes, did you push your changes ?

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataIntegrationTest.java (hunk 2 above)
Comment (shivajihmcts) [reply src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataIntegrationTest.java:62]: Its integration test case and data is coming from database table we can't achieve this in integration/functional test case

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java (hunk 3 above)
Comment (shivajihmcts) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataController.java:70]: done

File: src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java (hunk 1 above)
Comment (shivajihmcts) [reply src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/StaffRefDataFunctionalTest.java]: done

**Review by rajkatla-hmcts — APPROVED** [review]
